.github/workflows/check.yml

```diff
@@ -67,6 +67,95 @@ jobs:
             rust-version:  ${{ needs.get-rust-versions.outputs.stable }}
             cache: true
 
+    check-pretty-printers:
+        needs: [ calculate-git-info, get-rust-versions ]
+        strategy:
+            # `fromJSON` is used here to convert string output to boolean
+            fail-fast: ${{ fromJSON(needs.calculate-git-info.outputs.is_bors_branch) }}
+            matrix:
+                # TODO: check on mac-os as well
+                os: [ ubuntu-latest, windows-latest ]
+                # `fromJSON` is used here to convert string output to sequence.
+                # `matrix-version` is a string with list of stable + nightly versions.
+                # Make sequence from two outputs with version is not possible here.
+                # TODO: check on more rustc versions
+                rust-version: ${{ fromJSON(needs.get-rust-versions.outputs.matrix) }}
+                platform-version: [ 223, 231 ]
+                include:
+                    - os: ubuntu-latest
+                      rust-version: ${{ needs.get-rust-versions.outputs.old }}
+                      platform-version: 223
+        runs-on: ${{ matrix.os }}
+        env:
+            ORG_GRADLE_PROJECT_platformVersion: ${{ matrix.platform-version }}
+            ORG_GRADLE_PROJECT_compileNativeCode: false
+            RUST_BACKTRACE: 1
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Set up Rust
+              uses: actions-rs/toolchain@v1
+              with:
+                  profile: minimal
+                  toolchain: ${{ matrix.rust-version }}
+                  components: rust-src, rustfmt, clippy
+                  default: true
+
+            - name: Set up additional env variables
+              if: matrix.rust-version == needs.get-rust-versions.outputs.old
+                # see https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
+              run: |
+                  echo "ORG_GRADLE_PROJECT_ideaVersion=IU-2022.3" >> $GITHUB_ENV
+                  echo "ORG_GRADLE_PROJECT_clionVersion=CL-2022.3" >> $GITHUB_ENV
+                  echo "ORG_GRADLE_PROJECT_nativeDebugPluginVersion=223.7571.139" >> $GITHUB_ENV
```

Comment:
It seems it's worth to extract this (versions or the whole step) since we use the same code 2 times

Reply:
I've just extracted them to global environment variables